### PR TITLE
Feature integration tests

### PR DESCRIPTION
**Pull request by UbdaNam**
## What does this PR do?

This PR is raised to implement the following features: 👍 :

- [x] Add integration tests for all pages
- [x] Fix N+1 query performance issue
- [x] Use GitFlow to introduce new features
- [x] Setup rubocop and stylelint linters

## Description of the tasks completed 👇 

- [x] Add integration tests and Fix N+1 problem
- [x] Install rails gem
- [x] Use the linter tool that analyzes my source code to flag programming errors, bugs, stylistic errors, and suspicious constructs. 💯 
  
## How should it be tested

Clone the repository
```
git clone https://github.com/UbdaNam/Blogiverse.git
```
Navigate to the folder
```
cd Blogiverse
```
Checkout the branch
```
git checkout feature_integration_tests
```
Run migrations
```
rails db:migrate
```
Run tests
```
rspec spec/integration
```
BEFORE FIX

Before adding solution

AFTER FIX

After fixing N + 1 performance issue

### CODE REVIEW

File: spec/integration/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      click_link @third_post.text
+      expect(current_path).to eq(user_post_path(@first_user, @third_post))
+    end
+
+    it "Clicking on see all posts button will redirect to user's posts index page" do
+      visit user_path(@first_user)
+      click_link('See all posts')
+      expect(current_path).to eq(user_posts_path(@first_user))
+    end
+  end
```
Comment (Gambit142) [inline spec/integration/users_spec.rb]: - [x] Based on the requirements you ought to create a _**spec**_ file that is dedicated to testing the features of the application for the `Users#Show` page. Currently, you have one spec file for both the `Users#Index` page and the `Users#Show` page. Kindly put the test codes for the `Users#Show` page in a separate file. 👍 💻

File: spec/integration/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      expect(page).to have_content(@first_post.author.name)
+      expect(page).to have_content(@first_post.comments_counter)
+      expect(page).to have_content(@first_post.likes_counter)
+      expect(page).to have_content(@first_post.text)
+      expect(page).to have_content(@first_comment.author.name)
+      expect(page).to have_content(@second_comment.author.name)
+      expect(page).to have_content(@first_comment.text)
+      expect(page).to have_content(@second_comment.text)
+    end
+  end
```
Comment (Gambit142) [inline spec/integration/posts_spec.rb]: - [x] Based on the requirements you ought to create a _**spec**_ file that is dedicated to testing the features of the application for the `Posts#Show` page. Currently, you have one spec file for both the `Posts#Index` page and the `Posts#Show` page. Kindly put the test codes for the `Posts#Show` page in a separate file. 👍 💻

File: spec/integration/posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      expect(page).to have_css("img[src='#{@first_user.photo}']")
+      expect(page).to have_content(@first_user.posts_counter)
+      expect(page).to have_content(@first_post.text)
+      expect(page).to have_content(@second_post.text)
+      expect(page).to have_content(@third_post.text)
+      expect(page).to have_content(@first_comment.text)
+      expect(page).to have_content(@first_post.comments_counter)
+      expect(page).to have_content(@first_post.likes_counter)
+      expect(page).to have_selector('.pagination')
+    end
```
Comment (Gambit142) [inline spec/integration/posts_spec.rb]: - [x] For the `Posts#Index` page, you forgot to write a test that confirms the application renders **_a post's title_** of a user when a user visits the `Posts#Index` page. Kindly write this test. 👍 😎 

Screenshot_2

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @UbdaNam @Atifzada 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Use of **_Rspec testing library_** ✔️ 
- Use of _**Capybara**_ gem for writing integration tests ✔️ 
- Wrote feature tests for User and Post pages ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: spec/integration/posts_spec.rb (hunk 3 above)
Comment (UbdaNam) [reply spec/integration/posts_spec.rb]: Thanks for reviewing my PR 😀

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @UbdaNam and @Atifzada 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by UbdaNam** [comment]
Thank you @tresorsawasawa for reviewing my PR

**Pull request merged**